block/submit.go

```diff
@@ -282,7 +282,14 @@ func (m *Manager) UpdateLastSubmittedHeight(event pubsub.Message) {
 		return
 	}
 	h := eventData.EndHeight
-	if m.LastSubmittedHeight.Load() < h {
-		m.LastSubmittedHeight.Store(h)
+
+	for {
```

Comment:
maybe have similar syntax to 
```
func (m *Manager) UpdateTargetHeight(h uint64) {
	for {
		currentHeight := m.TargetHeight.Load()
		if m.TargetHeight.CompareAndSwap(currentHeight, max(currentHeight, h)) {
			break
		}
	}
}
```
using `max` instead of explicit 
```
if h <= curr {
	break
}
```

Reply:
done 👍